Using multiple reporters results into wrong exit code if test fails

**Summary:**
Using multiple reporters with  Lab v.15.2.0 -> wrong exit code when test fails
Using single reporter with Lab v.15.2.0 -> test doesn't run
Using no reporters -> test fail with correct exit code -> OK

See below for detailed logs:
-----------------------------------------------------------------------------
This works ok with Lab version **14.3.2:**

unit.js:
```
const { expect } = require('code');
const { it } = exports.lab = require('lab').script();

it('returns true when 1 + 1 equals 2', () => {

  expect(1 + 1).to.equal(3);
});

```

```
lab --verbose unit.js -r console -o stdout -r html -o coverage.html -r lcov -o lcov.info -r json -o data.json

✖ 1) returns true when 1 + 1 equals 2

Failed tests:

  1) returns true when 1 + 1 equals 2:
      actual expected
      23
      Expected 2 to equal specified value: 3
      at it (/unit.js:6:122

1 of 1 tests failed
Test duration: 8 ms
Assertions count: 0 (verbosity: 0.00)
No global variable leaks detected
Coverage: 100.00%

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! unit@8.0.0 test1: `lab --verbose unit.js -r console -o stdout -r html -o coverage.html -r lcov -o lcov.info -r json -o data.json`
npm ERR! Exit status 1
```
---------------------------------------------------------
But when using latest 15.2.0 version it does not work as expected and gives wrong exit code:

```
lab --verbose unit.js -r console -o stdout -r html -o coverage.html -r lcov -o lcov.info -r json -o data.json

✖ 1) returns true when 1 + 1 equals 2

Failed tests:

  1) returns true when 1 + 1 equals 2:

      actual expected
      23
      Expected 2 to equal specified value: 3
      at it (/unit.js:6:122

1 of 1 tests failed
Test duration: 5 ms
No global variable leaks detected
Coverage: 100.00%

```

---------------------------------------------------------

Also having just one reporter it doesn't even run the tests:

```
lab --verbose unit.js -r console -o console.log

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! unit@8.0.0 test1: `lab --verbose unit.js -r console -o console.log`
npm ERR! Exit status 1

0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'run', 'test1' ]
2 info using npm@5.6.0
3 info using node@v8.9.4
4 verbose run-script [ 'pretest1', 'test1', 'posttest1' ]
5 info lifecycle unit@8.0.0~pretest1: unit@8.0.0
6 info lifecycle unit@8.0.0~test1: unit@8.0.0
7 verbose lifecycle unit@8.0.0~test1: unsafe-perm in lifecycle true
8 verbose lifecycle unit@8.0.0~test1: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/username/unit/node_modules/.bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games
9 verbose lifecycle unit@8.0.0~test1: CWD: /home/username/unit
10 silly lifecycle unit@8.0.0~test1: Args: [ '-c', 'lab --verbose unit.js -r console -o console.log' ]
11 silly lifecycle unit@8.0.0~test1: Returned: code: 1  signal: null
12 info lifecycle unit@8.0.0~test1: Failed to exec test1 script
13 verbose stack Error: unit@8.0.0 test1: `lab --verbose unit.js -r console -o console.log`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:285:16)
13 verbose stack     at emitTwo (events.js:126:13)
13 verbose stack     at EventEmitter.emit (events.js:214:7)
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at emitTwo (events.js:126:13)
13 verbose stack     at ChildProcess.emit (events.js:214:7)
13 verbose stack     at maybeClose (internal/child_process.js:925:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
14 verbose pkgid unit@8.0.0
15 verbose cwd /home/username/unit
16 verbose Linux 4.9.0-5-amd64
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "test1"
18 verbose node v8.9.4
19 verbose npm  v5.6.0
20 error code ELIFECYCLE
21 error errno 1
22 error unit@8.0.0 test1: `lab --verbose unit.js -r console -o console.log`
22 error Exit status 1
23 error Failed at the unit@8.0.0 test1 script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]

```

---------------------------------------------------------

And if I don't give any reporters, it works just fine

```
lab --verbose unit.js

✖ 1) returns true when 1 + 1 equals 2

Failed tests:

  1) returns true when 1 + 1 equals 2:
      actual expected
      23
      Expected 2 to equal specified value: 3
      at it (/unit.js:6:20

1 of 1 tests failed
Test duration: 7 ms
No global variable leaks detected

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! unit@8.0.0 test1: `lab --verbose unit.js`
npm ERR! Exit status 1

```

Please let me know if you need further information

--- a/test/reporters.js
+++ b/test/reporters.js
@@ -1759,6 +1759,87 @@ describe('Reporter', () => {
             Fs.unlinkSync(filename);
         });
 
+        it('has correct exit code when test fails', async () => {
+
+            const Recorder = function () {
+
+                Stream.Writable.call(this);
+
+                this.content = '';
+            };
+
+            Hoek.inherits(Recorder, Stream.Writable);
+
+            Recorder.prototype._write = function (chunk, encoding, next) {
+
+                this.content += chunk.toString();
+                next();
+            };
+
+            const Test = require('./coverage/basic');
+            const script = Lab.script();
+            script.experiment('test', () => {
+
+                script.test('succeeds', () => {
+
+                    Test.method(true);
+                });
+
+                script.test('fails', () => {
+
+                    return Promise.reject('error');
+                });
+            });
+
+            const consoleRecorder = new Recorder();
+            const htmlRecorder = new Recorder();
+            const filename = Path.join(Os.tmpdir(), [Date.now(), process.pid, Crypto.randomBytes(8).toString('hex')].join('-'));
+
+            const { code, output } = await Lab.report(script, { reporter: ['lcov', 'html', 'console'], output: [filename, htmlRecorder, consoleRecorder], coverage: true, coveragePath: Path.join(__dirname, './coverage/basic.js') });
+            expect(code.console).to.equal(1);
+            expect(code.lcov).to.equal(1);
+            expect(code.html).to.equal(1);
+            expect(output.lcov).to.equal(Fs.readFileSync(filename).toString());
+            expect(output.html).to.equal(htmlRecorder.content);
+            expect(consoleRecorder.content).to.contain('Coverage: 100.00%');
+
+            Fs.unlinkSync(filename);
+        });
+
+        it('can run a single reporter', async () => {
+
+            const Recorder = function () {
+
+                Stream.Writable.call(this);
+
+                this.content = '';
+            };
+
+            Hoek.inherits(Recorder, Stream.Writable);
+
+            Recorder.prototype._write = function (chunk, encoding, next) {
+
+                this.content += chunk.toString();
+                next();
+            };
+
+            const Test = require('./coverage/basic');
+            const script = Lab.script();
+            script.experiment('test', () => {
+
+                script.test('succeeds', () => {
+
+                    Test.method(true);
+                });
+            });
+
+            const consoleRecorder = new Recorder();
+
+            const { code, output } = await Lab.report(script, { reporter: ['console'], output: [consoleRecorder], verbose: true, coverage: true, coveragePath: Path.join(__dirname, './coverage/basic.js') });
+            expect(code).to.equal(0);
+            expect(consoleRecorder.content).to.contain('Coverage: 100.00%');
+            expect(consoleRecorder.content).to.equal(output);
+        });
 
         it('that are duplicates with multiple outputs are supported', async () => {
 

